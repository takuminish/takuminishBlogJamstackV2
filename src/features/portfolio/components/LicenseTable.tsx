import TableContainer from "@mui/material/TableContainer";
import { License } from "../../../pages";
import Table from "@mui/material/Table";
import TableHead from "@mui/material/TableHead";
import TableRow from "@mui/material/TableRow";
import React from "react";
import TableCell from "@mui/material/TableCell";
import { format } from "date-fns";
import TableBody from "@mui/material/TableBody";
import Paper from "@mui/material/Paper";

type Props = {
  licenses: License[];
};

function LicenseTable(props: Props) {
  const { licenses } = props;

  return (
    <TableContainer component={Paper}>
      <Table>
        <TableHead>
          <TableRow>
            <TableCell>資格名</TableCell>
            <TableCell>取得年</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {licenses.map((license) => (
            <TableRow
              key={license.fieldId}
              sx={{ "&:last-child td, &:last-child th": { border: 0 } }}
            >
              <TableCell component="th" scope="row">
                {license.name}
              </TableCell>
              <TableCell align="right">
                {format(new Date(license.passedAt), "yyyy")}
              </TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </TableContainer>
  );
}

export default LicenseTable;
