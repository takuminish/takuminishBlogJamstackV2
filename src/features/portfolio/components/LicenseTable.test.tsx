import React from "react";
import { render } from "@testing-library/react";
import LicenseTable, { Props } from "./LicenseTable";

const createDOM = () => {
  const props: Props = {
    licenses: [
      {
        fieldId: "license",
        name: "license1",
        passedAt: new Date("2022-06-23T15:00:00.000Z"),
        isITLicense: true,
      },
      {
        fieldId: "license",
        name: "license2",
        passedAt: new Date("2022-06-23T15:00:00.000Z"),
        isITLicense: true,
      },
    ],
  };
  return render(<LicenseTable licenses={props.licenses} />);
};

test("snapshot", () => {
  const { container } = createDOM();

  expect(container).toMatchSnapshot();
});
