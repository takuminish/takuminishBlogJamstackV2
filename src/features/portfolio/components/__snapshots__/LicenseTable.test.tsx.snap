// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
  >
    <table
      class="MuiTable-root css-rqglhn-MuiTable-root"
    >
      <thead
        class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          >
            資格名
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          >
            取得年
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root css-34nofg-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            scope="row"
          >
            license1
          </th>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
          >
            2022
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-34nofg-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            scope="row"
          >
            license2
          </th>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
          >
            2022
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
